test/adapter_test.py

```diff
@@ -314,6 +315,43 @@ def test_non_object_response(self):
       ),
     )
 
+  @mock.patch('iota.adapter.request')
+  def test_timeout(self, request_mock):
+    # create dummy response
+    request_mock.return_value = mock.Mock(text='{ "dummy": "payload"}', status_code=200)
+
+    # create adapter
+    mock_payload = {'dummy': 'payload'}
+    adapter = HttpAdapter('http://localhost:14265')
+
+    # test with default timeout
+    adapter.send_request(payload=mock_payload)
+    _, kwargs = request_mock.call_args
+    self.assertEqual(kwargs['timeout'], socket.getdefaulttimeout())
+
+    request_mock.request_mock()
+
+    # test with explicit attribute
+    adapter.timeout = 77
+    adapter.send_request(payload=mock_payload)
+    _, kwargs = request_mock.call_args
+    self.assertEqual(kwargs['timeout'], 77)
+
+    request_mock.request_mock()
+
+    # test with timeout in kwargs
+    adapter.send_request(payload=mock_payload, timeout=88)
+    _, kwargs = request_mock.call_args
+    self.assertEqual(kwargs['timeout'], 88)
+
+    request_mock.request_mock()
+
+    # test with timeout at adapter creation
+    adapter = HttpAdapter('http://localhost:14265', timeout=99)
+    adapter.send_request(payload=mock_payload)
+    _, kwargs = request_mock.call_args
+    self.assertEqual(kwargs['timeout'], 99)
```

Comment:
Could we split these into separate tests?  That will ensure that each of the test scenarios run, even if one of the assertions fails — avoiding the situation where you fix the code to make an assertion pass... only for the very next assertion in the test to fail.

Reply:
done

Reply:
Thanks!

Reply:
Could we give each of these tests a different name, to make it easier to tell what exactly is being tested in each scenario?

Also, should lines 364-368 also be refactored into their own test?